ebos/eclgenericproblem.hh

```diff
@@ -382,8 +382,22 @@ protected:
         using GridType = std::remove_cv_t< typename std::remove_reference<decltype(gridView_.grid())>::type>;
         return lookUpData_.template getFieldPropIdx<GridType>(elemIdx);
     }
+    
+    std::vector<double> getAssignDoubleOnLeaf(const FieldPropsManager& fieldPropManager,
+                                              const std::string& propString, const unsigned int& numElems) const
+    {
+        return lookUpData_.assignFieldPropsDoubleOnLeaf(fieldPropManager, propString, numElems);
+    }
+
+    std::vector<unsigned int> getAssignIntOnLeaf(const FieldPropsManager& fieldPropManager,
+                                                 const std::string& propString,
+                                                 const unsigned int& numElems, bool needsTranslation) const
+    {
+        return lookUpData_.template assignFieldPropsIntOnLeaf<unsigned int>(fieldPropManager, propString, numElems,
+                                                                            needsTranslation);
+    }
```

Comment:
Wouldn't it be possibel to skip these functions and do the necessary calls in the lambdas below (you are capturing this there), e.g.:
```c++
auto getDoubleOnLeaf = [this](const FieldPropsManager& fieldPropManager, const std::string& propString,
                              const unsigned int& numElems)
{
       return this->lookUpData_.assignFieldPropsDoubleOnLeaf(fieldPropManager, propString, numElems);
};
```
Or am I missing something (again)?

Reply:
EclGenericPloblem has the member lookupdata_ and it's a derived class of EclProblem. I tried to downcast "this( EclProblem)" to be able to get the lookupdata memeber, and it didn't work. (I was probably doing something wrong). 
I will check out your suggestion, thanks!

Reply:
Then I really missed something. I always forget about this complicated inheritance that we have. I might need some time to think about this then.

Reply:
There is multiple [inheritance](https://github.com/OPM/opm-simulators/blob/df0daa1a1656c73d9e71be4a80a027edfd907a80/ebos/eclproblem.hh#L110-L113) . Maybe that was the problem. Have you tried casting to [BaseType](https://github.com/OPM/opm-simulators/blob/df0daa1a1656c73d9e71be4a80a027edfd907a80/ebos/eclproblem.hh#L115-L118)?
```
static_cast<BaseType* const>(this)->lookUpData_....
```

Reply:
I probably placed `const` wrong. :sweat_smile:

Reply:
It might work, but also need a friend declaration, or some extra method returning the data member lookupdata_, since it's declared protected. I'll check it out

Reply:
Shouldn't it be accessible in the derived class if it is protected in the base class? 
We should try to prevent adding friend declarations here.

Reply:
Many thanks for your feedback, it seems to work. I'll run jenkins again

Reply:
If protected members of a base class (here EclGenericProblem) are static, then any friends or member functions of the derived class (here EclProblem) can access them. If it's not static, only through pointer, reference, or object of the derived class.  
So, you're right, it should work without that extra method getLookUpData(). I'll double-check.

In any case, here EclProblem is publically derived from EclGenericProblem, and the static_cast is "upcast" (I used the wrong term before: downcast). So, here, if I get it correctly, the

Reply:
I think I see the issue now. Once we static_cast<BaseType*> "this", we are actually upcasting, so now we are a base class thing wanting to access protected member of a derived class (and not the other way around). That could explain why we need the extra public member function returning lookUpData_ in the base class EclGenericProblem

Reply:
:flushed: I think the problem was me giving advice that is nonsense. I should have thought this through. My apologies.

In the lambda, which has its own class with no connection to the EclGenericProblem class and its parent or derived classes, we cannot access anything of EclGenericProblem that is not public.

I have sent you an email with a possible way.

Reply:
Sent for the fix-idea! Incorporated in the last version